### PR TITLE
 New UI and initial scientific notation

### PR DESCRIPTION
**Pull request by neoJINXD**
This PR has the new UI implemented and hooked up to the server for it to run the calculations with the server.
Kept the old version in the files if we ever needed it again, can be accessed by adding '/old.html' to the URL.

Could not get the color for which mode is selected working so for now the text of the selected on is bolded. Will check if i can add color to it.

Also contains the first version for the scientific notation conversion, still missing it's unit tests but will implement that later on a seperate branch.

### CODE REVIEW

**Comment by SwatiPareek** [comment]
Looks really good. 
One consideration point for us (perhaps later) in terms of user-friendliness is that if a user first encounters this and wants to type in values, its not obvious they need to press above the display. My gut instinct would be to press on the display (where the output is).  Could we set the cursor there or would it be better to write input/output? Or do you think since the user can press buttons, they'll eventually realize?

**Comment by SwatiPareek** [comment]
Also do we still have the clear memory function still working? How can we access memory with the new UI?

**Comment by VisusAdAstra** [comment]
√(3) = 2.0??? and help button doesn't work

File: functions/scientific.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        exp = len(whole_part[1:])
+        output = whole_part[0] + '.' + whole_part[1:] + decimal_part + 'e+' + str(exp)
+    return output
+
+
+print(convert_to_sn(0.0056))
+print(convert_to_sn(0.00000856))
+print(convert_to_sn(0.075056))
+print(convert_to_sn(750.56))
+print(convert_to_sn(7506235474))
```
Comment (VisusAdAstra) [inline functions/scientific.py:36]: remove test line plz

**Comment by VisusAdAstra** [comment]
Evaluation Error: mod is not a recognized symbol.

**Comment by Emancus09** [comment]
> √(3) = 2.0??? and help button doesn't work

This is super weird. 3^0.5 works fine and sqrt(x) works well with other values of x but 3 gives 2. I'll look into this.

**Comment by Emancus09** [comment]
Can we make the input field clear whenever a value is calculated again? Was that removed on purpose or by accident?

**Comment by neoJINXD** [comment]
> Looks really good.
> One consideration point for us (perhaps later) in terms of user-friendliness is that if a user first encounters this and wants to type in values, its not obvious they need to press above the display. My gut instinct would be to press on the display (where the output is). Could we set the cursor there or would it be better to write input/output? Or do you think since the user can press buttons, they'll eventually realize?
@SwatiPareek 

So do you think it would be better if we set the input text box to be focused on the page load, to show a typing cursor in the input field?

> Also do we still have the clear memory function still working? How can we access memory with the new UI?
@SwatiPareek 

From my usage, the memory doesnt kick in until a first input is given. After that, a dropdown indicator shows up on the right side of the input field with the list of previously entered equations. If I recall, a similar thing happened when I was using the old ui. The CLRM button will clear and empty the list of option, making the dropdown disappear. 

> √(3) = 2.0??? and help button doesn't work
@VisusAdAstra 

Help button wasnt hooked up since when I made the PR we didnt have the help/user guide made, Will add it before merging and after talking to Tara on how I should connect it.

> Evaluation Error: mod is not a recognized symbol.
@VisusAdAstra 

Had the mod button input the wrong thing in the input field. If we are putting in mod as in the modulus operator %, should be fixed with latest commit.

> Can we make the input field clear whenever a value is calculated again? Was that removed on purpose or by accident?
@Emancus09

This was done intentionally to leave an indication of what what the equation that was last submitted when receiving the answer. I dont think there's another good place we can put that information.

**Comment by VisusAdAstra** [comment]
Is √(3) = 2.0 the only remain problem? I approve pr rn, you can merge after fixing it

**Pull request merged**